walkthrough/utils.py

```diff
@@ -134,3 +136,107 @@ def generate_step(cookiecutter_path: Path, data: Dict[str, Any]) -> Path:
         )
 
     return DATASET_DIR
+
+
+def _check_env() -> None:
+    """Check if environment variables are set correctly."""
+    po.put_markdown("""## Checking `.env` file...""")
+
+    ok = True
+    for env_name in ("GRAPHER_USER_ID", "DB_USER", "DB_NAME", "DB_HOST"):
+        if getattr(config, env_name) is None:
+            ok = False
+            po.put_warning(
+                po.put_markdown(f"Environment variable `{env_name}` not found, do you have it in your `.env` file?")
+            )
+
+    if ok:
+        po.put_success(po.put_markdown("`.env` configured correctly"))
+
+
+def _show_environment() -> None:
+    """Show environment variables."""
+    po.put_markdown("## Environment")
+    po.put_info(
+        po.put_markdown(
+            f"""
+    * **GRAPHER_USER_ID**: `{config.GRAPHER_USER_ID}`
+    * **DB_USER**: `{config.DB_USER}`
+    * **DB_NAME**: `{config.DB_NAME}`
+    * **DB_HOST**: `{config.DB_HOST}`
+    """
+        )
+    )
+    if OWIDEnv().env_type_id == OWIDEnvType.LIVE:
+        po.put_warning(
+            po.put_markdown(
+                "You are currently connected to the **live** database! If you are not sure what this means, please"
+                " consult the with ETL team."
+            )
+        )
+
+
+class OWIDEnvType(Enum):
+    LIVE = 1
+    STAGING = 2
+    LOCAL = 3
+    UNKNOWN = -1
+
+
+class OWIDEnv:
+    env_type_id: Literal[OWIDEnvType.LIVE, OWIDEnvType.STAGING, OWIDEnvType.LOCAL, OWIDEnvType.UNKNOWN]
+
+    def __init__(
+        self,
+        env_type_id: Union[
+            Literal[OWIDEnvType.LIVE, OWIDEnvType.STAGING, OWIDEnvType.LOCAL, OWIDEnvType.UNKNOWN], None
+        ] = None,
+    ):
+        if env_type_id is None:
+            self.env_type_id = self.get_env_type_id_from_config()
+        else:
+            self.env_type_id = env_type_id
+
+    def get_env_type_id_from_config(
```

Comment:
Would rename this `detect_env_type()`, since the config isn't listing the type, instead it's being inferred from magic values of config strings.